qabelbox/src/main/java/de/qabel/qabelbox/notifications/QblNotificationManager.java

```diff
@@ -0,0 +1,57 @@
+package de.qabel.qabelbox.notifications;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.Nullable;
+import android.support.v7.app.NotificationCompat;
+
+public abstract class QblNotificationManager {
+
+    private final NotificationManager notificationManager;
+    private final Context context;
+
+    private int idSequence = 0;
+
+    public QblNotificationManager(Context context) {
+        this.context = context;
+        this.notificationManager = (NotificationManager)
+                context.getSystemService(Context.NOTIFICATION_SERVICE);
+    }
+
+    protected abstract String getTag();
+
+    protected int generateId() {
+        return idSequence++;
+    }
+
+    protected Context getContext() {
+        return this.context;
+    }
+
+    protected void notify(int id, Notification notification) {
+        notificationManager.notify(getTag(), id, notification);
+    }
+
+    protected NotificationCompat.Builder createNotification(Intent intent, String contentTitle, int iconRes, @Nullable String contentText, @Nullable Integer progress) {
```

Comment:
If this part would be extracted to a separate and specialized class, there would be no need for the "if (progress != null)".